src/lighthouse.rs

```diff
@@ -90,55 +113,83 @@ fn quorum_changed(a: &Vec<QuorumMember>, b: &Vec<QuorumMember>) -> bool {
 }
 
 // Checks whether the quorum is valid and an explanation for the state.
-fn quorum_valid(state: &RoomState, opt: &LighthouseOpt) -> (bool, String) {
-    let mut first_joined = Instant::now();
+fn quorum_valid(
+    now: Instant,
+    heartbeats: &HashMap<String, Instant>,
+    state: &RoomState,
+    opt: &LighthouseOpt,
+) -> (bool, String) {
+    let mut first_joined = now;
+
+    let healthy_participants: HashMap<String, QuorumMemberDetails> = state
+        .participants
+        .clone()
+        .into_iter()
+        .filter(|(replica_id, _details)| {
+            let last_heartbeat = heartbeats.get(replica_id);
+            if last_heartbeat.is_none() {
+                return false;
```

Comment:
Add an optional message?

Reply:
We print the "heathly" replica count in the status message below, the dashboard should make this pretty obvious if no replicas have heartbeated